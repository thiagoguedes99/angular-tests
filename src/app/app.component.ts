import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterLink, RouterLinkActive, RouterOutlet } from '@angular/router';

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [CommonModule, RouterOutlet, RouterLink, RouterLinkActive],
  // templateUrl: './app.component.html',
  template: `
    <h1>Hello, {{ title }}</h1>

    <button
      [routerLink]="['/basic/counter']"
      routerLinkActive="router-link-active"
      [routerLinkActiveOptions]="{exact: true}"
    >
      baasic counter
    </button>

    <button
      [routerLink]="['/basic/counter/30']"
      routerLinkActive="router-link-active"
      [routerLinkActiveOptions]="{exact: true}"
    >
      baasic counter param
    </button>

    <button
      [routerLink]="['/basic/pokemmon']"
      routerLinkActive="router-link-active"
    >
      baasic request pokemon
    </button>

    <button
      [routerLink]="['/basic/family']"
      routerLinkActive="router-link-active"
    >
      baasic input output father son
    </button>


    <router-outlet></router-outlet>
  `,
  styleUrl: './app.component.scss'
})
export class AppComponent {
  title = 'angular-tests';
}
