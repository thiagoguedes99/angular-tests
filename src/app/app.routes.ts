import { Routes } from '@angular/router';
import { CounterComponent } from './basic/counter/counter.component';
import { PokemonComponent } from './basic/pokemon/pokemon.component';
import { FatherComponent } from './basic/family/father/father.component';

export const routes: Routes = [
  {
    path: 'basic/counter',
    component: CounterComponent
  },
  {
    path: 'basic/pokemmon',
    component: PokemonComponent
  },
  {
    path: 'basic/family',
    component: FatherComponent
  },
  // {
  //   path: '**',
  //   redirectTo: 'basic/family'
  // }

];
