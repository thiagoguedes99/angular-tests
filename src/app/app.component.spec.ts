import { TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
// import { ActivatedRoute, convertToParamMap } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { of } from 'rxjs';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent, RouterTestingModule],
      // providers: [ActivatedRoute]
      // providers: [
      //   {
      //     provide: ActivatedRoute,
      //     useValue: {
      //       queryParams: of(convertToParamMap({
      //         initial: ""
      //       }))
      //     }
      //   }
      // ]
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it('should to snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.nativeElement as HTMLElement;
    fixture.detectChanges();

    expect(component).toMatchSnapshot();
  })

  it(`should have the 'angular-tests' title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('angular-tests');
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled.querySelector('h1')?.textContent).toContain('Hello, angular-tests');
  });
});
