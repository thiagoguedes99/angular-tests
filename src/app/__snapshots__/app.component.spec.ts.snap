// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should to snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Hello, angular-tests
  </h1>
  <button
    routerlinkactive="router-link-active"
    tabindex="0"
  >
     baasic counter 
  </button>
  <button
    routerlinkactive="router-link-active"
    tabindex="0"
  >
     baasic counter param 
  </button>
  <button
    routerlinkactive="router-link-active"
    tabindex="0"
  >
     baasic request pokemon 
  </button>
  <button
    routerlinkactive="router-link-active"
    tabindex="0"
  >
     baasic input output father son 
  </button>
  <router-outlet />
  
</div>
`;
